### ISSUE TITLE
SAC W/O AMP overperforms than W AMP

### ISSUE THREAD

**Issue by mzhao-conti** [jiamiya/HOPE#12]
The ablation experiment result (Table V in paper) shows that SAC W AMP performs better than W/O AMP
![Image](https://github.com/user-attachments/assets/79781828-f297-464c-bd40-dd7e57317bc2)

In the code, the default configuration for SAC agent is W/O AMP. In case choose_action is set to True for SAC agent, the evaluation result shows more steps taken for a successful parking. Is there something misunderstood?
```
eval_mix_scene.py
    choose_action = True if isinstance(rl_agent, PPO) else False
    with torch.no_grad():
        # eval on extreme
        env.set_level('Extrem')
        log_path = save_path+'/extreme'
        if not os.path.exists(log_path):
            os.makedirs(log_path)
        eval(env, parking_agent, episode=eval_episode, log_path=log_path, post_proc_action=choose_action)
```

**Comment by jiamiya** [comment]
Yes, you are right about this. In practice we find that SAC works not that well with Action Mask Post-processing when inference, probably due to the way the algorithm handle the action distribution. Nevertheless, avoiding collision in training is crucial, without which the performance decline greatly.

**Comment by mzhao-conti** [comment]
Regarding Action Mask, below are my understandings, please correct me in case:

action_mask.py
```
def init_vehicle_box(self,):
      ...
      actions = np.array(self.action_space)
      ...
      delta_phi = 0.5*actions[:,1]/10/radius
      ...
```
Is here 0.5 the time in simulator per one interaction(in configs.py I found STEP_LENGHT is 5e-2 and NUM_STEP is 10), and 10 the K times to discretize velocity(according to the equation 12 in paper)? 

Another confusion is that actions[:,1] here actually is 1, that means the max step speed is 1, why it's not VALID_SPEED[1]?
Because in car_parking_base.py, the action space tells the speed ranges from VALID_SPEED[0] to VALID_SPEED[1]; and in vehicle.py, KSModel use the speed sampled from that range to calculate the movement.

After read more, it shows that only the 1st action is sampled from VALID_SPEED[0] to VALID_SPEED[1]. RSPlanner and agent always take 1 as the max step speed. The confusion is that KSModel use VALID_SPEED[1] 2.5 as max speed to maneuver the vehicle in the gym env. And this impacts the path plan.

i.e. set VALID_SPEED[1] 2.5

![Image](https://github.com/user-attachments/assets/c2382a54-69bb-410e-a09f-212469f05d00)

i.e. set VALID_SPEED[1] 1

![Image](https://github.com/user-attachments/assets/80130241-e2ba-4f6f-bc73-c1e8d7d10877)

**Comment by jiamiya** [comment]
Thank you for your detailed questions and observations. 

1. Regarding the 0.5 and 10 in the calculation: Yes, you're correct on both counts. The 0.5 represents the time per interaction (from STEP_LENGTH and NUM_STEP), and the 10 corresponds to the K value used for velocity discretization as in equation 12 of the paper.

2. About actions[:,1] being 1 rather than VALID_SPEED[1]: This was primarily an engineering design choice. The implementation could alternatively use the maximum speed magnitude or RL output mean to fill the masked action's second dimension. While we haven't thoroughly investigated how this choice affects results, you can try some experiments on this aspect if interested in.

3. Concerning the RSPlanner using 1 while KSModel uses VALID_SPEED[1] (2.5): These values are actually consistent because the RSPlanner's output is normalized (where 1 represents maximum step size), which is then denormalized in the env_wrapper to the actual maximum speed (2.5).
